test/space_test.jl

```diff
@@ -203,12 +203,14 @@ end
     @test node_neighbors((2, 2), gridspace) == [(2, 1), (1, 2), (3, 2), (2, 3)]
     @test node_neighbors((1, 1), gridspace) == [(2, 1), (1, 2)]
     a = add_agent!((2, 2), gridspace, rand())
-    add_agent!((3, 2), gridspace, rand())
+    b = add_agent!((3, 2), gridspace, rand())
     @test space_neighbors((1, 2), gridspace) == [1]
     @test sort!(space_neighbors((1, 2), gridspace, 2)) == [1, 2]
     @test_throws MethodError space_neighbors((1, 2), gridspace, 1.5)
     @test sort!(space_neighbors((2, 2), gridspace)) == [1, 2]
     @test space_neighbors(a, gridspace) == [2]
+    @test get_node_agents(node_neighbors(a, gridspace), gridspace) == [[], [], [b], []]
+    @test get_node_agents(node_neighbors(b, gridspace), gridspace) == [[], [a], []]
```

Comment:
This is confusing. How do I receive 4 directions for `a`, but only 3 directions for `b`?

Also, since this is explicitly returning directional data, there should be some tests when the space has `moore = true`, as well as a verification that `periodic = true` finds neighbors correctly.

Reply:
> How do I receive 4 directions for `a`, but only 3 directions for `b`?

Because the grid is 3x3 and agent b is at the edge of the grid which returns 3 neighboring nodes.

Sure, we can also test for moore neighorhood.

Reply:
Ah. I see. I will amend my comment above to reflect this.